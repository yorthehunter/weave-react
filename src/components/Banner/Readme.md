Banner types
```js
<React.Fragment>
  <Banner visible={true} type="info">Hello</Banner>
  <Banner visible={true} type="warning">Hello</Banner>
  <Banner visible={true} type="error">Hello</Banner>
  <Banner visible={true} type="success">Hello</Banner>
  <Banner visible={true}>Hello</Banner>
</React.Fragment>
```

Banner using `content` prop
```js
<Banner type="info" visible={true} content="Hello" />
```

Flat Banner
```js
<Banner type="info" visible={true} flat content="Hello" />
```

Render HTML inside Banner
```js
<Banner type="info" visible={true} html="<b>Hello</b>" />
```

Non-dismissable Banner
```js
<Banner visible={true} type="success" dismissable={false}>Hello</Banner>
```
