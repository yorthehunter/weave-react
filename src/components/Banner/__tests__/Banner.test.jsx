import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Banner from '../';

describe('Banner renders correctly', () => {
  it('when visible with default values', () => {
    const rendered = renderer.create(<Banner visible />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('with content passed as a prop', () => {
    const wrapper = shallow(<Banner visible content="Here's some content!" />);
    const rendered = toJson(wrapper);
    expect(rendered).toMatchSnapshot();
  });

  it('with content passed as a child', () => {
    const rendered = renderer.create(<Banner visible>Test content</Banner>).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('with html content', () => {
    const rendered = renderer.create(<Banner visible html="<strong>Content!</strong>" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('as null if banner is not visible', () => {
    const bannerClass = '.weave-banner';
    const wrapper = mount(<Banner>Banner</Banner>);
    expect(wrapper.find(bannerClass)).toHaveLength(0);
  });
});

describe('Banner', () => {
  const bannerClass = '.weave-banner';
  const wrapper = mount(<Banner visible>Banner</Banner>);

  it('correctly handles banner type', () => {
    wrapper.setProps({ type: 'error' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--error')).toEqual(true);
    wrapper.setProps({ type: 'success' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--success')).toEqual(true);
    wrapper.setProps({ type: 'info' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--info')).toEqual(true);
    wrapper.setProps({ type: 'warning' });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--warning')).toEqual(true);
    wrapper.setProps({ type: null });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--error')).toEqual(false);
  });

  it('is responsive', () => {
    wrapper.setProps({ responsive: true });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--responsive')).toEqual(true);
  });

  it('is flat', () => {
    wrapper.setProps({ flat: true });
    expect(wrapper.find(bannerClass).hasClass('weave-banner--flat')).toEqual(true);
  });

  it('removes itself when close button clicked', () => {
    expect(wrapper.find(bannerClass)).toHaveLength(1);
    wrapper.find('button').simulate('click');
    expect(wrapper.find(bannerClass)).toHaveLength(0);
  });
});
