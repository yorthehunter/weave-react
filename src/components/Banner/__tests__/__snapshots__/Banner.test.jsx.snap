// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly when visible with default values 1`] = `
<div
  className="weave-banner"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
  />
  <button
    className="weave-banner__close"
    onClick={[Function]}
  >
    <svg
      height={24}
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="var(--font-color)"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;

exports[`Banner renders correctly with content passed as a child 1`] = `
<div
  className="weave-banner"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
  >
    Test content
  </p>
  <button
    className="weave-banner__close"
    onClick={[Function]}
  >
    <svg
      height={24}
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="var(--font-color)"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;

exports[`Banner renders correctly with content passed as a prop 1`] = `
<CustomTag
  className={null}
  componentClasses={
    Array [
      "weave-banner",
      "",
      "",
      false,
      false,
    ]
  }
  componentRef={null}
  componentStyles={Object {}}
  content="Here's some content!"
  dismissable={true}
  flat={false}
  href={null}
  html={null}
  onClick={null}
  responsive={false}
  size={null}
  style={Object {}}
  tag="div"
  type={null}
  value={null}
  visible={true}
>
  <p
    className="weave-banner__message"
  >
    Here's some content!
  </p>
  <button
    className="weave-banner__close"
    onClick={[Function]}
  >
    <Icon
      color="var(--font-color)"
      height={24}
      name="close"
      width={24}
    />
  </button>
</CustomTag>
`;

exports[`Banner renders correctly with html content 1`] = `
<div
  className="weave-banner"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <p
    className="weave-banner__message"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<strong>Content!</strong>",
      }
    }
  />
  <button
    className="weave-banner__close"
    onClick={[Function]}
  >
    <svg
      height={24}
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close
      </title>
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
        fill="var(--font-color)"
      />
      <polygon
        fillOpacity="0"
        points="0 0 24 0 24 24 0 24"
      />
    </svg>
  </button>
</div>
`;
